README.md

```diff
@@ -562,6 +565,39 @@ Accesses a notice's payload, which can be read or filtered. Payload includes:
 notice[:params][:my_param] = 'foobar'
 ```
 
+### Promise
+
+`Airbrake::Promise` represents a simplified promise object (similar to promises
+found in JavaScript), which allows chaining callbacks that are executed when
+the promise is either resolved or rejected.
+
+#### Promise#then
+
+Attaches a callback to be executed when a promise is resolved (fulfilled). The
+promise is resolved whenever the Airbrake API successfully accepts your
+exception.
+
+Yields notice id (the id of an error at Airbrake, discoverable by
+visiting http://airbrake.io/locate/{{notice_id}}) and returns `self`.
+
+```rb
+Airbrake.notify('Oops').then { |notice_id| puts notice_id }
+#=> 000543c7-95ac-06ba-edd6-e81084f4f50b
+```
```

Comment:
There is always a stupid solution ... but better get this merged then adding more stuff ...
```Ruby
def value
  value = nil
  Thread.new { promise.then { |e| value = e || false } }
  sleep 0.01 if value.nil?
  value
end
```